main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala

```diff
@@ -684,22 +684,27 @@ object Weeder2 {
     }
 
     private def pickEqualityConstraints(tree: Tree): Validation[List[EqualityConstraint], CompilationMessage] = {
-      val maybeContraintList = tryPick(TreeKind.Decl.EqualityConstraintList, tree)
-      val constraints = traverseOpt(maybeContraintList)(t => {
+      val maybeConstraintList = tryPick(TreeKind.Decl.EqualityConstraintList, tree)
+      val constraints = traverseOpt(maybeConstraintList)(t => {
         val constraintTrees = pickAll(TreeKind.Decl.EqualityConstraintFragment, t)
         traverse(constraintTrees)(visitEqualityConstraint)
       })
 
-      mapN(constraints)(_.getOrElse(List.empty))
+      mapN(constraints) {
+        case optListOpt =>
+          optListOpt.getOrElse(List.empty)
```

Comment:
Uuuuh... optListOption? What do you need to return? Perhaps flatten can be used?

Reply:
or collect